### DISCUSSION TITLE
Better *initial* global search UI

### DISCUSSION THREAD

**Discussion by KittyBorgX** [helix-editor/helix#4610] [Ideas] [answered]
The global search, when invoked by `SPACE + e` looks very similar to the normal search apart from the `global-search` text in front of the slash. To make the user experience better, here is my suggestion: 

Making a telescope like menu, similar to `Open file picker` (`SPACE + f`) in helix and keep the searching option there. This looks similar to Find Files of Telescope in Neovim. 

Implementation : I'm willing to take this up and open a pr if this idea gets upvoted / approved :)

**Comment by archseer** [comment]
> Making a telescope like menu, similar to `Open file picker` (`SPACE + f`) in helix and keep the searching option there. This looks similar to Find Files of Telescope in Neovim.


Can you explain the difference? I don't understand, given that I've never used telescope.nvim.

**Reply by KittyBorgX** [reply]
Sure! 

At the moment, it [helix] looks somewhat like this : 
![image](https://user-images.githubusercontent.com/82328083/200178326-6ebaf504-57f9-4b3b-8334-29b884c9dcc2.png)

My proposal is to make it look something like this: 
![image](https://user-images.githubusercontent.com/82328083/200178467-4b8f35e3-6e02-4f50-93cb-3139af071e4c.png)

The major difference is incremental updating of search results. The implementation idea of mine was to copy the code for the File picker in helix (`SPACE + f`) and use it for the global search. It looks nicer and feels more user friendly :) 

Edit : To give a bit more clarity, helix does show it in a similar fashion as shown in the below image, however, it displays this only after the user presses `ENTER` after searching for the phrase through `SPACE + /`

![image](https://user-images.githubusercontent.com/82328083/200178731-2810b9a6-f0c9-4604-9fc5-75d7edea1a12.png)

My suggestion was to make the entire process of searching consistent like how I mentioned above

**Reply by the-mikedavis** [reply]
This sounds like #196, no?

**Reply by KittyBorgX** [reply]
Yep! 😅 
I forgot to check for the same in the open issues since I only searched for it in the discussions and closed issues. My bad!

**Comment by KittyBorgX** [comment, marked as answer]
> This sounds like #196, no?

Yep, Duplicate of #196, my bad!
